modules/database/variables.tf

```diff
@@ -0,0 +1,59 @@
+variable "namespace" {
+  type        = string
+  description = "Friendly name prefix used for tagging and naming Azure resources."
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of the resource group in which to create the database."
+}
+
+variable "location" {
+  type        = string
+  description = "Specifies the supported Azure location where the resource exists."
+}
+
+variable "sku_name" {
+  type        = string
+  default     = "GP_Standard_D4ds_v4"
+  description = "Specifies the SKU Name for this MySQL Server"
+}
+
+variable "database_subnet_id" {
+  type        = string
+  description = "Network subnet id for database"
+}
+
+variable "database_private_dns_zone_id" {
+  type        = string
+  description = "The identity of the private DNS zone in which the database will be deployed."
+}
+
+
+variable "database_version" {
+  description = "Version for MYSQL"
+  type        = string
+  default     = "8.0.21"
+
+  validation {
+    condition     = contains(["5.7", "8.0.21"], var.database_version)
+    error_message = "We only support MySQL: \"5.7\"; \"8.0.21\"."
+  }
+}
+
+variable "database_availability_mode" {
+  description = ""
+  type        = string
+  default     = "SameZone"
```

Comment:
I imagine we want the default to be "ZoneRedundant".  When ever possible we want our infra to weather a Zone going down.  We won't support multi-region, but our official terraform should adhere to the cloud contract which says we should assume a single zone can just go away.  Again, not necessarily required for MVP, but we should definitely think about and work towards multi zone operations for MySQL, k8s, and Redis

Reply:
With ZoneRedundant, the limitation we might have is the regions it's currently available at. There's very few regions that [support ZoneRedundant today](https://docs.microsoft.com/en-us/azure/mysql/flexible-server/overview#azure-regions). In the US, it's only `East US` and `South Central US` only. If customers want their deployment in other regions we might have to default back to SameZone if not the deployment would fail.